### PR TITLE
github: add pr templates and commit-message-lint

### PR DESCRIPTION
**Pull request by qiankehan**
For the details please check the commits msgs of these two patches.
And please add [commit-message-lint app](https://github.com/apps/commit-message-lint) to tp-libvirt when merging this PR. @chunfuwen

### CODE REVIEW

**Comment by chunfuwen** [comment]
I wonder whether we can put those templates under docs folder(which can be newly created) since .git hold quite sensitive data with version control. See some reference, https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository

**Comment by qiankehan** [comment]
> Good initiative!
> I wonder whether we can put those templates under docs folder(which can be newly created) since .git hold quite sensitive data with version control. See some reference, https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository


The pr templates and the commit-message-lint configuration are under **.github** not **.git**. I think it's ok here

**Comment by chunfuwen** [comment]
> > Good initiative!
> > I wonder whether we can put those templates under docs folder(which can be newly created) since .git hold quite sensitive data with version control. See some reference, https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository
> 
> The pr templates and the commit-message-lint configuration are under **.github** not **.git**. I think it's ok here

I agree. I wonder whether you can give some simple steps to describe how can we use template when do git commit

**Comment by qiankehan** [comment]
> > > Good initiative!
> > > I wonder whether we can put those templates under docs folder(which can be newly created) since .git hold quite sensitive data with version control. See some reference, https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/creating-a-pull-request-template-for-your-repository
> > 
> > 
> > The pr templates and the commit-message-lint configuration are under **.github** not **.git**. I think it's ok here
> 
> I agree. I wonder whether you can give some simple steps to describe how can we use template when do git commit

See https://medium.com/better-programming/github-templates-the-smarter-way-to-formalize-pull-requests-among-development-teams-89f8d6a204f
Pr templates provide some suggestions for the **format of PR to the contributors**.
For the commit message lint, it will check if the title of commit message matches **PATTERN: SUMMARY** format. This format will help improve the readability of the commits.

File: .github/PULL_REQUEST_TEMPLATE/bug_fix.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+# <Message body>
+**_# Use short but accurate sentence to describe: what the issue is; how you fix
+it; why it should be fixed_**
+Reference: **_Paste the reference link if any background of this issue_**
+
+Test results:
+**_# Provide test results of these involved cases after applying the PR_**
+
+Signed-off:
```
Comment (chunfuwen) [inline .github/PULL_REQUEST_TEMPLATE/bug_fix.md:12]: The part may be not needed, since it will be automatically enforced when do git commit in any cases

**Comment by qiankehan** [comment]
On Tue, Sep 29, 2020 at 11:29 AM Chunfu Wen <notifications@github.com>
wrote:

> *@chunfuwen* commented on this pull request.
> ------------------------------
>
> In .github/PULL_REQUEST_TEMPLATE/bug_fix.md
> <https://github.com/autotest/tp-libvirt/pull/3098#discussion_r496356752>:
>
> > @@ -0,0 +1,13 @@
> +# <Title>
> +PATTERN: SUBJECT
> +
> +# <Message body>
> +**_# Use short but accurate sentence to describe: what the issue is; how you fix
> +it; why it should be fixed_**
> +Reference: **_Paste the reference link if any background of this issue_**
> +
> +Test results:
> +**_# Provide test results of these involved cases after applying the PR_**
> +
> +Signed-off:
>
> The part may be not needed, since it will be automatically enforced when
> do git commit in any cases
>
Yes. The check for signed-off is better to be done by the the **commit
lint**

File: .github/PULL_REQUEST_TEMPLATE/bug_fix.md (hunk 1 above)
Comment (qiankehan) [reply .github/PULL_REQUEST_TEMPLATE/bug_fix.md:12]: Yes. The check for signed-off is better to be done by the the **commit lint**. I will update it

File: .github/PULL_REQUEST_TEMPLATE/bug_fix.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# <Title>
+PATTERN: SUBJECT
+
+# <Message body>
+**_# Use short but accurate sentence to describe: what the issue is; how you fix
+it; why it should be fixed_**
+Reference: **_Paste the reference link if any background of this issue_**
+
+Test results:
```
Comment (chloerh) [inline .github/PULL_REQUEST_TEMPLATE/bug_fix.md:9]: Are we supposed to put test result into commit message?

Comment (kylazhang) [reply .github/PULL_REQUEST_TEMPLATE/bug_fix.md:9]: I think he means the Pull requests message

File: .github/PULL_REQUEST_TEMPLATE/tiny_fix.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# <Title>
+PATTERN: SUBJECT
+
+Is it the following type of patch series:
+- **_# Comment fixes_**
+- **_# Typo fixes_**
+- **_# Document fixes_**
+
+Signed-off:
+**_Signed-off-by: YOUR_NAME <YOUR_EMAIL>_**
```
Comment (kylazhang) [inline .github/PULL_REQUEST_TEMPLATE/tiny_fix.md:10]: I think result also needed for tiny fix

File: .github/PULL_REQUEST_TEMPLATE/bug_fix.md (hunk 2 above)
Comment (kylazhang) [reply .github/PULL_REQUEST_TEMPLATE/bug_fix.md:9]: I think it is not suitable to add test results in commit message

Comment (chloerh) [reply .github/PULL_REQUEST_TEMPLATE/bug_fix.md:9]: Agreed

Comment (qiankehan) [reply .github/PULL_REQUEST_TEMPLATE/bug_fix.md:9]: Yes. The test results should be mentioned in PR msgs not commit msgs.

File: .github/PULL_REQUEST_TEMPLATE/bug_fix.md (hunk 1 above)
Comment (qiankehan) [reply .github/PULL_REQUEST_TEMPLATE/bug_fix.md]: Signed-off is removed from PR template

**Comment by qiankehan** [comment]
v2 Update:
- Removed Signed-off-by checking from PR templates.
- Add error messages and Signed-off-by checking in Commit Message Lint.

**Comment by qiankehan** [comment]
Update:
1. My repo worked well with commit-message-lint
 installed. See the [results of commit message lint run](https://github.com/qiankehan/tp-libvirt/pull/1/checks?check_run_id=1346535069)
2. pr templates doesn't work on my repo. I will check why
3. For forced signed-off, it cannot be realized in this PR. Please update the [repo settings](https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/enabling-required-commit-signing) to enforce it.

**Pull request closed without merging**